code/modules/surgery/organs/stomach/external/_external_organs.dm

```diff
@@ -0,0 +1,219 @@
+	var/preference = ""
+	///Sprite datums we use to draw on the bodypart
+	var/list/sprite_datums = list()
+	///Key of the icon states of all the sprite_datums for easy caching
+	var/cache_key = ""
+
+	///Reference to the limb we're inside of
+	var/obj/item/bodypart/ownerlimb
+
+/**_mob_sprite is optional if you havent set sprite_datums for the object, and is used mostly to generate sprite_datums from a persons DNA
+* For _mob_sprite we make a distinction between "Round Snout" and "round". Round Snout is the name of the sprite datum, while "round" would be part of the sprite
+* I'm sorry
+*/
+/obj/item/organ/external/Initialize(mapload, _mob_sprite)
+	. = ..()
+
+	if(_mob_sprite)
+		add_sprite_datum(_mob_sprite)
+
+	generate_icon_cache()
+
+/obj/item/organ/external/Insert(mob/living/carbon/reciever, special, drop_if_replaced)
+	var/obj/item/bodypart/limb = reciever.get_bodypart(zone)
+
+	if(!limb)
+		return FALSE
+
+	limb.external_organs.Add(src)
+	ownerlimb = limb
+
+	. =  ..()
+
+	limb.contents.Add(src)
+
+	reciever.update_body_parts()
+
+/obj/item/organ/external/Remove(mob/living/carbon/organ_owner, special)
+	. = ..()
+
+	if(ownerlimb)
+		ownerlimb.external_organs.Remove(src)
+		ownerlimb.contents.Remove(src)
+
+	organ_owner.update_body_parts()
+
+/obj/item/organ/external/transfer_to_limb(obj/item/bodypart/bodypart, mob/living/carbon/bodypart_owner)
+	. = ..()
+
+	bodypart.external_organs.Add(src)
+	bodypart.contents.Add(src)
+
+///Use a sprite NAME (Beautiful Moth Butt Wings) and find the appropriate sprite_accessory datum
+/obj/item/organ/external/proc/add_sprite_datum(base_icon, clear_all=FALSE)
+	if(clear_all)
+		sprite_datums.Cut()
+	var/sprite_datum = get_sprite_datum(base_icon)
+	if(!sprite_datum)
+		CRASH("[type] could not find a sprite accessory datum with the display name '[base_icon]")
+
+	sprite_datums.Add(sprite_datum)
+
+///Add the overlays we need to draw on a person. Called from _bodyparts.dm
+/obj/item/organ/external/proc/get_overlays(list/overlay_list, _dir, _layer, body_type, _color)
+	generate_icon_cache()
+	for(var/datum/sprite_accessory/sprite_datum as anything in sprite_datums)
+		var/g = (body_type == FEMALE) ? "f" : "m"
+		var/finished_icon_state = (sprite_datum.gender_specific ? g : "m") + "_" + preference + "_" + sprite_datum.icon_state + mutant_bodyparts_layertext(_layer)
+		var/mutable_appearance/appearance = mutable_appearance(sprite_datum.icon, finished_icon_state, layer = -_layer)
+		appearance.dir = _dir
+
+		if(sprite_datum.color_src) //There are multiple flags, but only one is ever used so meh :/
+			appearance.color = _color
+
+		if(sprite_datum.center)
+			center_image(appearance, sprite_datum.dimension_x, sprite_datum.dimension_y)
+
+		overlay_list += appearance
+
+///Generate a unique key based on our sprites. So that if we've aleady drawn these sprites, they can be found in the cache and wont have to be drawn again (blessing and curse)
+/obj/item/organ/external/proc/generate_icon_cache()
+	cache_key = ""
```

Comment:
I don't really like this proc just mutating itself, for the sake of cleanliness make this a pure function that only returns a value, then do `cache_key = generate_icon_cache()`.